packages/rulesets/src/spectral/az-arm.ts

```diff
@@ -731,6 +732,21 @@ const ruleset: any = {
       },
     },
 
+    // RPC Code: RPC-Put-V1-30
+    TagsAreTopLevelPropertiesOnly: {
+      rpcGuidelineCode: "RPC-Put-V1-30",
+      description: "Tags must be defined as a top-level property, not in the properties bag.",
+      severity: "error",
```

Comment:
Tejaswi, the intention for this rule is to not allow tags for proxy resources at all even as a top level property. So the documentation, rule name and error messages need to reflect that - "Tags should not be specified in the properties bag for proxy resources. Consider using a Tracked resource instead."